core/src/main/java/org/elasticsearch/common/xcontent/ConstructingObjectParser.java

```diff
@@ -295,6 +295,10 @@ public Value parse(XContentParser parser, Context context) throws IOException {
         }
     }
 
+    public String getName() {
```

Comment:
Since this is implemented also in ObjectParser, I would make this an abstract method in AbstractObjectParser. I think this way ValuesSourceParserHelper.declareFields() could work AbstractObjectParser without all the current duclication. Not 100% sure though but I'd give it a try.